File: udp-server/src/main/java/io/sparkled/udpserver/impl/command/RequestCommand.java
package io.sparkled.udpserver.impl.command;

import io.sparkled.model.setting.SettingsCache;
import io.sparkled.music.PlaybackState;
import io.sparkled.udpserver.impl.RequestHandlerImpl;

public abstract class RequestCommand {

    public abstract byte[] getResponse(String[] args, SettingsCache settings, PlaybackState playbackState);

    byte[] getErrorResponse() {
        return RequestHandlerImpl.ERROR_CODE_BYTES;
    }
}

File: udp-server/src/main/kotlin/io/sparkled/udpserver/impl/command/RequestCommand.kt
package io.sparkled.udpserver.impl.command

import io.sparkled.model.setting.SettingsCache
import io.sparkled.music.PlaybackState

abstract class RequestCommand {

    abstract fun getResponse(args: List<String>, settings: SettingsCache, playbackState: PlaybackState): ByteArray

    internal val emptyResponse: ByteArray = ByteArray(0)
}
